### ISSUE TITLE
Break up model, again

### ISSUE THREAD

**Issue by jfroelich** [jfroelich/rss-reader#766]
Each iteration and back and forth I learn a few things, and gain what I think is a better insight into the design I want.

In the previous overhaul of the database functionality, I merged everything into a single file, I renamed things, and I exposed the functionality as methods of a `Model` object.

This is not bad. However, I liked many aspects of the prior implementation.

There are benefits to using a single object. It is easy to import. There are fewer files. There is no need to pass around the same common parameters (such as the database connection) to each helper.

However, there are negatives. `model.js` is monolithic. It feels difficult to edit. It does not feel cohesive. While these operations are all related to the database, they are in many cases unrelated to each other.

I can provide that a feature is a part of a model simply by using a folder to group together model-related features. I do not need to roll that up into a class.

By breaking things up, the caller includes only what they want. If a higher level module depends on only one or two operations from the model layer, it couples directly to those operations, instead of the entire thing. The module still exists in the model layer, it is just accessed directly instead of through a single entry point.

Just how much more complexity is there if I use 100 modules instead of one class with 100 methods? I am not sure there is actually a complexity change. I don't buy into this rationale.

I really like how compartmentalized the model operation modules become. I can focus on one thing at a time. Compared to looking at 100 unrelated functions all bunched together in the same class, it makes it easier to edit.

Some of the model functions need help. They have helper functions. I want them private but not nested. In some cases I created a ton of non-operation methods in the same place as the operational methods to provide the utility. I just hate this. When using separate modules, however, I can freely include any number of 'private' helper methods right there in the module. There is no need to get tricky with visibility.

Some modules have only one simple function, if I am consistent and do this for each model method, but that is ok. It is ok to have a ton of modules. The concern is not performance. The concern is maintainability. I really need to overcome this idea that a module should not exist unless it has enough complexity. Some rooms of a building are just simple closets, but they are nevertheless considered rooms just like the big main ones.

I do not think that reducing some parameters that happen to be shared is alone sufficient justification to switch to an object-oriented design. I think this was the red herring I followed in the prior design. So what if several of the model operation functions accept a connection parameter. This does not mean they should be a part of the model.

While many different modules within the model layer now all require intimate knowledge of the database structure, I think it is fine. These modules are in the model layer. We can state that any module in the model layer has intimate knowledge of the storage schema. If we need to make a change to the schema, it will be roughly the same level of difficulty to update the functionality spread out through a few files as it would be to investigate every method of a very large single file mega-class.

The bottom line is ease of maintenance. I think the multi-module design is just easier to maintain.

**Comment by jfroelich** [comment]
Later, separate out conn and channel. It was a nice experiment but I think these are separate.

**Comment by jfroelich** [comment]
Minor remaining work:

* revise as ops: sanitize-entry, sanitize-feed, validate-entry, validate-feed
